### DISCUSSION TITLE
To use DRF endpoint

### DISCUSSION THREAD

**Discussion by kon-m** [vercel/community#4685] [Help] [answered]
### Summary

Hi,

I made my app by React & Typescript. I deploy it in vercel.
This app use the API endpoint from Django REST Framework.

I set CORS in Django settings as follows.

CORS_ALLOWED_ORIGINS = [
"vercel origins",
]

but my vercel app says messages as follows.
my vercel app can't use DRF API endpoint.
Django app says no error messages.

Access to fetch at 'DRF' from origin 'vercel origins' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

I use fetch fuction in Vercel app as follows.

fetch("DRF")

I have written 'corsheaders', and 'corsheaders.middleware.CorsMiddleware' in setting.py of Django.
Does Vercel app need other settings to use DRF API endpoint ?
If you know about this, Please let me know.

Thank you so much.

Regards.

### Steps to Reproduce

Please make the API by Django REST Framework.
And you use it on fetch function in React + Typescript.
And you deploy it from Github.
Please set up to use DRF in settings.py

**Comment by amyegan** [comment, marked as answer]
This guide should help you set the necessary headers: https://vercel.com/guides/how-to-enable-cors

**Comment by kon-m** [comment]
Hi,

Thank you so much.

Regards.

2023年11月2日(木) 2:22 Amy ***@***.***>:
